### PR TITLE
[24_5] tm2html tool based on libmogan

### PR DESCRIPTION
**Pull request by Charonxin**
## Why you open this Pull Request?

cc: https://github.com/XmacsLabs/mogan/pull/1056

## What work have you done in the current Pull Request?

- [x] convert from single file to file
- [x] convert from dir to dir
- [x] remove useless code of other options in tm2html.cpp

### CODE REVIEW

**Comment by Charonxin** [comment]
try:

```
TEXMACS_PATH=/home/charon/mogan/TeXmacs xmake run tm2html -headless -c devel/release_1.2.0.tm html-out/test.html -q
```
convert file to file

```
TEXMACS_PATH=/home/charon/mogan/TeXmacs  xmake run tm2html -headless -dir devel html-out -q
```
convert dir to dir

**Comment by darcy-shen** [comment]
Remove the link `TeXmacs/plugins`, and I guess ci will work fine.

**Comment by Charonxin** [comment]
> Remove the link `TeXmacs/plugins`, and I guess ci will work fine.

ingenious calculation

File: TeXmacs/progs/doc/tmweb.scm [hunk 1]
```diff
@@ -113,6 +113,9 @@
 	      (url->list u4))))
 
 (tm-define (tmweb-convert-dir tm-dir html-dir)
+  (display* "[DEBUG] enter scheme func\n")
```
Comment (darcy-shen) [inline TeXmacs/progs/doc/tmweb.scm:116]: Remove debugging code

File: xmake.lua [hunk 2]
```diff
@@ -1114,6 +1114,101 @@ for _, filepath in ipairs(os.files("TeXmacs/tests/*.scm")) do
...
+        })
+    elseif is_plat("linux") then
+        add_installfiles({
+            "misc/scripts/mogan"
+        })
+    end
+end
+
+target ("tm2html") do
+    if not is_plat("wasm") then
```
Comment (darcy-shen) [inline xmake.lua:1206]:
``` lua
if not is_plat("wasm") then
  target ("tm2html") do
  end
end
```

Comment (darcy-shen) [reply xmake.lua:1206]: 这样更好，wasm里面不需要出现这个 target

**Comment by Charonxin** [comment]
Can I add `TeXmacs/plugins` to `.gitignore` ?

**Comment by darcy-shen** [comment]
> Can I add TeXmacs/plugins to .gitignore ?

A good idea but not necessary. Because I planned to move `plugins` to `TeXmacs/plugins`

**Comment by darcy-shen** [comment]
See https://github.com/XmacsLabs/mogan/pull/587 

Let us adjust the plugin folder position after the merge of #1078

**Review by darcy-shen — APPROVED** [review]
LGTM

**Comment by Charonxin** [comment]
单车都蹬到教室了，ci还没跑完 :sweat_smile:

**Comment by darcy-shen** [comment]
ci有点问题，需要调整一下，这块后面交给 @paradisuman ，让他熟悉一下ci

**Pull request merged**